lib/resources/aws/aws_config_delivery_channel.rb

```diff
@@ -0,0 +1,67 @@
+class AwsConfigurationDeliveryChannel < Inspec.resource(1)
+  name 'aws_config_delivery_channel'
+  desc 'Verifies settings for AWS Configuration Delivery Channel'
+  example "
+    describe aws_config_delivery_channel do
+      it { should exist }
+      it { should be_default }
+      its('s3_bucket_name') { should_not be_nil }
+      its('sns_topic_arn') { should_not be_nil }
+    end
+  "
+  supports platform: 'aws'
+
+  include AwsSingularResourceMixin
+  attr_reader :channel_name, :s3_bucket_name, :s3_key_prefix, :sns_topic_arn,
+              :delivery_frequency
+
+  def to_s
+    "Configuration_Delivery_Channel: #{@channel_name}"
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:channel_name],
+      allowed_scalar_name: :channel_name,
+      allowed_scalar_type: String,
+    )
+
+    # Make sure channel_name is given as param
+    if validated_params[:channel_name].nil?
+      raise ArgumentError, 'You must provide a channel_name to aws_config_delivery_channel'
+    end
+
+    validated_params
+  end
+
+  def fetch_from_api
+    backend = BackendFactory.create(inspec_runner)
+    query = { delivery_channel_names: [@channel_name] }
+    @resp = backend.describe_delivery_channels(query)
```

Comment:
FYI, per #2636 , we should now wrap every call we make to AWS in a block, like this:
```
catch_aws_errors do
  @resp = backend.describe_delivery_channels(query)
end
```
If your API call indicates a search miss by throwing an exception, as some do, you are still obliged to catch it.  catch_aws_errors is intended for more general things - bad creds, insufficient perms, etc.

Reply:
Still need to wrap the API call in the error-catcher.